internal/kafka/internal/services/quota/ams_quota_service_test.go

```diff
@@ -15,6 +15,133 @@ import (
+		ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Build()
+		return ca, nil
+	},
+	GetOrganisationIdFromExternalIdFunc: func(externalId string) (string, error) {
+		return fmt.Sprintf("fake-org-id-%s", externalId), nil
+	},
+	GetQuotaCostsForProductFunc: func(organizationID, resourceName, product string) ([]*v1.QuotaCost, error) {
+		if product != string(ocm.RHOSAKProduct) {
+			return []*v1.QuotaCost{}, nil
+		}
+		rrbq1 := v1.NewRelatedResource().
+			BillingModel(string(v1.BillingModelStandard)).
+			Product(string(ocm.RHOSAKProduct)).
+			ResourceName(resourceName).
+			Cost(1)
+
+		qcb, err := v1.NewQuotaCost().Allowed(1).Consumed(0).OrganizationID(organizationID).RelatedResources(rrbq1).
+			CloudAccounts(
+				v1.NewCloudAccount().CloudProviderID("aws").CloudAccountID("1234567890"),
+				v1.NewCloudAccount().CloudProviderID("ibm").CloudAccountID("2345678901"),
+				v1.NewCloudAccount().CloudProviderID("azure").CloudAccountID("1234567890"),
+			).
+			Build()
+		if err != nil {
+			panic("unexpected error")
+		}
+
+		return []*v1.QuotaCost{qcb}, nil
+	},
+}
+
+func Test_AMSValidateBillingAccount(t *testing.T) {
+	type fields struct {
+		ocmClient   ocm.Client
+		kafkaConfig *config.KafkaConfig
+	}
+	type args struct {
+		orgId            string
+		billingAccountId string
+		marketplace      *string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "check passes when requested billing account is found in account list",
+			args: args{
+				"test",
+				"1234567890",
+				&[]string{"aws"}[0],
+			},
+			fields: fields{
+				ocmClient:   ocmClientMockWithCloudAccounts,
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr: false,
+		},
+		{
+			name: "check passes when marketplace is not provided and account id is unambiguous",
+			args: args{
+				"test",
+				"2345678901",
+				nil,
+			},
+			fields: fields{
+				ocmClient:   ocmClientMockWithCloudAccounts,
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr: false,
+		},
+		{
+			name: "check fails when billing account id is not found",
+			args: args{
+				"test",
+				"xxx",
+				nil,
+			},
+			fields: fields{
+				ocmClient:   ocmClientMockWithCloudAccounts,
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr: true,
+		},
+		{
+			name: "check fails when billing account id is found in the wrong marketplace",
+			args: args{
+				"test",
+				"2345678901",
+				&[]string{"aws"}[0],
+			},
+			fields: fields{
+				ocmClient:   ocmClientMockWithCloudAccounts,
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr: true,
+		},
+		{
+			name: "check fails when billing account id ambiguous",
+			args: args{
+				"test",
+				"1234567890",
+				nil,
+			},
+			fields: fields{
+				ocmClient:   ocmClientMockWithCloudAccounts,
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gomega.RegisterTestingT(t)
```

Comment:
this could be moved before the for loop and `gomega.` could be removed, if the import line looks like this: `. "github.com/onsi/gomega"`